src/test/groovy/org/prebid/server/functional/tests/GeoSpec.groovy

```diff
@@ -0,0 +1,212 @@
+package org.prebid.server.functional.tests
+
+import org.prebid.server.functional.model.config.AccountAuctionConfig
+import org.prebid.server.functional.model.config.AccountConfig
+import org.prebid.server.functional.model.db.Account
+import org.prebid.server.functional.model.pricefloors.Country
+import org.prebid.server.functional.model.pricefloors.PublicCountryIpV4
+import org.prebid.server.functional.model.request.auction.BidRequest
+import org.prebid.server.functional.model.config.AccountSetting
+import org.prebid.server.functional.model.request.auction.Device
+import org.prebid.server.functional.model.request.auction.Geo
+import org.prebid.server.functional.util.PBSUtils
+import java.time.Instant
+
+import static org.prebid.server.functional.model.AccountStatus.ACTIVE
+
+import static org.prebid.server.functional.model.request.auction.TraceLevel.VERBOSE
+import static org.prebid.server.functional.util.privacy.model.State.ALABAMA
+import static org.prebid.server.functional.util.privacy.model.State.ONTARIO
+
+class GeoSpec extends BaseSpec {
+
+    private static final String GEO_LOCATION_REQUESTS = "geolocation_requests"
+    private static final String GEO_LOCATION_FAIL = "geolocation_fail"
+    private static final String GEO_LOCATION_SUCCESSFUL = "geolocation_successful"
+    private static final Map<String, String> GEO_LOCATION = ["geolocation.type"                               : "configuration",
+                                                             "geolocation.configurations.[0].address-pattern" : PublicCountryIpV4.USA.ipV4,
+                                                             "geolocation.configurations.[0].geo-info.country": Country.USA.ISOAlpha2,
+                                                             "geolocation.configurations.[0].geo-info.region" : ALABAMA.abbreviation]
+
+    def "PBS should populate geo with country and region when geo location enabled in host and account config"() {
+        given: "Default basic generic BidRequest"
+        def config = AccountConfig.defaultAccountConfig.tap {
+            settings = new AccountSetting(geoLookup: defaultAccountGeoLookup)
+        }
+        def defaultPbsService = pbsServiceFactory.getService(
+                ["settings.default-account-config": encode(config),
+                 "geolocation.enabled"            : "true"] + GEO_LOCATION)
+        def bidRequest = BidRequest.defaultBidRequest.tap {
```

Comment:
given: "PBS service with geolocation and default account configs"
        def config = AccountConfig.defaultAccountConfig.tap {
            settings = new AccountSetting(geoLookup: defaultAccountGeoLookup)
        }
        def defaultPbsService = pbsServiceFactory.getService(
                ["settings.default-account-config": encode(config),
                 "geolocation.enabled"            : "true"] + GEO_LOCATION)
        and: "Generic BidRequest with geo-data" 
        def bidRequest = BidRequest.defaultBidRequest.tap {